src/runtime/thread_storage_scope.h

```diff
@@ -96,6 +98,8 @@ struct StorageScope {
         return "global" + tag;
       case StorageRank::kShared:
         return "shared" + tag;
+      case StorageRank::kDynShared:
```

Comment:
Let us use the same storage scope, but introduce a different tag. e.g. shared.dyn

Reply:
I changed dynamic shared memory to be a tagged memory, for `kShared`, but I had to add a check `e->scope.tag != ".dyn"` in a few places in `lower_device_storage_access_info.cc` and  `storage_rewrite.cc` to workaround error/segfault.